File: generate_table_versions.py
import json
import pandas as pd

input_file = 'minimum_gem_output.json'
with open(input_file, 'r') as file:
    data = json.load(file)

rows = []
for integration_name, versions in data.items():
    ruby_min, jruby_min = versions
    rows.append({"Integration": integration_name, "Ruby Min": ruby_min, "JRuby Min": jruby_min})

df = pd.DataFrame(rows)

output_file = 'integration_versions.md'

with open(output_file, 'w') as md_file:
    md_file.write("| Integration | Ruby Min | JRuby Min |\n")
    md_file.write("|-------------|-----------|----------|\n")
    for _, row in df.iterrows():
        md_file.write(f"| {row['Integration']} | {row['Ruby Min']} | {row['JRuby Min']} |\n")

File: .github/scripts/generate_table_versions.rb
require 'json'

input_file = 'gem_output.json'
output_file = 'integration_versions.md'

data = JSON.parse(File.read(input_file))

comment = "# Integrations\n\n"
header = "| Integration | Ruby Min |  Ruby Max | JRuby Min | JRuby Max |\n"
separator = "|-------------|----------|-----------|----------|----------|\n"
rows = data.map do |integration_name, versions|
    ruby_min, ruby_max, jruby_min, jruby_max = versions.map { |v| v || "None" }
    "| #{integration_name} | #{ruby_min} | #{ruby_max} | #{jruby_min} | #{jruby_max} |"
end

File.open(output_file, 'w') do |file|
  file.puts comment
  file.puts header
  file.puts separator
  rows.each { |row| file.puts row }
end
